### PR TITLE
Adding analysis

### PR DESCRIPTION
**Pull request by pwl**
added `run_analysis` and `poll_analysis_until_validated`.

### CODE REVIEW

File: NAMESPACE [hunk 1]
```diff
@@ -19,6 +19,8 @@ export(get_valid_matrix_formats)
 export(get_valid_technologies)
 export(get_workflow)
 export(get_workflows)
+export(run_analysis)
+export(poll_analysis_until_validated)
```
Comment (ChristianSauer) [inline NAMESPACE]: until_completed? seems to be a better name

File: R/FGResponse.R [hunk 2]
```diff
@@ -30,3 +30,40 @@ setMethod("show", "FGResponse", function(object) {
...
+        fgresponse <- new("FGErrorResponse",
+                          path=url, content = parsed, validation_errors=validation_errors)
+    }
+    else if(response["status_code"] == 400){
+        parsed <- jsonlite::fromJSON(httr::content(response, "text"), simplifyVector = FALSE)
+        message = parsed[["message"]]
+        error_code = parsed[["error_code"]]
+        warning(stringr::str_interp("Upload of dataset failed due to an error: ${message}"),
+             call. = FALSE)
+        fgresponse <- new("FGErrorResponse",
```
Comment (ChristianSauer) [inline R/FGResponse.R]: Is it always an FGErrrResponse? Some apis return an error object? FGErrorResponse.R vs FGValidationProblem.R

File: R/analysis.R [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+    headers <- get_default_headers(connection)
+    url <- paste(connection@base_url, "analysis/api/v1/analyses/compute", sep="")
+    body <- list(analysis_template_id = template@Id, name=title)
+
+    response <- httr::POST(url, headers, body=body, encode="json")
+    return(parse_response(response, "analysis"))
+}
+
+#' Waits for the validation of the dataset to complete.
```
Comment (ChristianSauer) [inline R/analysis.R]: Copy & paste .
Please use the correct words ;)

File: R/analysis.R [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+        if(n_tasks_finished < length(get_finished(parsed$tasks))){
+            tasks_finished = get_finished(parsed$tasks)
+            n_tasks_finished = n_tasks_finished + 1
+            message(task_to_text(tasks_finished[[n_tasks_finished]]))
+        }
+    }
+}
+
+task_to_text <- function(task){
```
Comment (ChristianSauer) [inline R/analysis.R]: use stringr:.str_interp`?

File: R/FGResponse.R (hunk 2 above)
Comment (pwl) [reply R/FGResponse.R]: Oh, I didn't notice the validation problem, shouldn't this also be used in the dataset context?

File: R/analysis.R (hunk 4 above)
Comment (pwl) [reply R/analysis.R]: I guess it's fine as it is

**Comment by pwl** [comment]
I added the tests for the seurat upload (indirectly though)

File: R/FGResponse.R (hunk 2 above)
Comment (ChristianSauer) [reply R/FGResponse.R]: Yes, datasets are also affected.

**Comment by ChristianSauer** [comment]
@pwl You can merge this, but we need then to open another pr to check that we got Error / ValidationProblem right.,

File: R/FGResponse.R (hunk 2 above)
Comment (pwl) [reply R/FGResponse.R]: I think I fixed that with the newst commit

**Pull request merged**